vendor/bat-native-ads/src/bat/ads/internal/tokens/refill_unblinded_tokens/refill_unblinded_tokens.cc

```diff
@@ -107,7 +107,33 @@ void RefillUnblindedTokens::Refill() {
 
   nonce_ = "";
 
-  RequestSignedTokens();
+  InitiateRequestSignedTokens();
+}
+
+void RefillUnblindedTokens::InitiateRequestSignedTokens() {
+  BLOG(1, "InitiateRequestSignedTokens");
+
+  auto captcha_callback =
+      std::bind(&RefillUnblindedTokens::OnGetScheduledCaptcha, this,
+                std::placeholders::_1);
+  AdsClientHelper::Get()->GetScheduledCaptcha(wallet_.id,
+                                              std::move(captcha_callback));
+}
+
+void RefillUnblindedTokens::OnGetScheduledCaptcha(
+    const std::string& captcha_id) {
+  BLOG(1, "OnGetScheduledCaptcha");
+
+  if (captcha_id.empty()) {
+    RequestSignedTokens();
```

Comment:
Can we add tests for both valid and empty `captcha_id`'s

Reply:
Yes, done.

Reply:
so, if there is some kind of server error, won't we effectively DDOS ourselves here? We're not passing back any information about what the error was so won't we just be in an endless loop of making requests?

Reply:
I don't think so. If the server returns no captcha id, we just request tokens as always.